profile/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+    };*/
+
+/*    let read_options = CsvReadOptions{
+        has_header:false,
+        schema: Some(Arc::new(schema)),
+        parse_options: Arc::new(parse_options),
+        ..Default::default()     
+    };*/
+    let lf= LazyCsvReader::new(path)
+    .with_separator(b' ')
+    .with_has_header(false)
+    .with_schema(Some(Arc::new(schema))).finish()?;
+    
+/*    let df = CsvReader::new(file)
+        .with_options (read_options)
+        .finish()?;
+*/
+    Ok(lf)
+
+}
+
+//function that appends doppler shift
+pub fn append_doppler_shift(df:LazyFrame)// I take ownership of the data frame since I will produce a new one and want the old one to be dropped after appending
+->LazyFrame{
+    let doppler_shift = lit(1.0) - col("velocity")/lit(CLIGHT)*lit(1.0e3);
+    let result =
+    df.clone().lazy().
+    with_column(doppler_shift.clone().alias("doppler shift"));
+    result
+}
+
+
+
+//function that returns the flux for each cell
+pub fn return_flux_for_cell_thetaphi(
+    coschi:f64,
+    temperature:f64,
+    log_gravity:f64,
+    doppler_shift:f64,
+    area:f64,
+    wavelengths:&[f64],
+    )->Vec<Vec<f64>>{
+	//function that selects 4 grids temperature +_ times gravity+_ ..and 
+    //a function that creates lazyframes for the intensity grids
+    //filter values that are outside of wavelength range
+    //construct a lambda array dopplershifted
+    //look for a function that select lambdas that are close to the values of the lambda array
+    let shifted_wavelengths = get_doppler_shifted_wavelengths(doppler_shift, wavelengths);
+    let filter_exp = filter_if_contains_wavelenght(&shifted_wavelengths, 
+        0.1); 
+    //function that appends Ic and I to each data frame
+	//function that linearly interpolates Ic and I from the 8 grids
+    //
+
+    vec![vec![0.0]]
+}
+
+fn get_doppler_shifted_wavelengths(doppler_shift:f64,
+        wavelengths:&[f64])->Vec<f64>{
+            let mut shifted:Vec<f64>=Vec::new();
+            for lambda in wavelengths.iter(){
+                shifted.push(*lambda);
+            }
+            shifted
+}
+
+fn filter_if_contains_wavelenght(shifted_wavelengths:&[f64],threshold:f64)->Option<Expr>{
+
+    let mut combined_filter_exp: Option<Expr> = None;
+
+    for wavelength in shifted_wavelengths.iter(){
+        let lower_bound = lit(*wavelength-threshold);
+        let upper_bound = lit(*wavelength + threshold);
+        
+        let current_filter_exp = col("wave_length")
+        .gt_eq(lower_bound)
+        .and(col("wave_length").lt_eq(upper_bound));
+
+        combined_filter_exp = match combined_filter_exp{
+            Some(expr) => {Some(expr.or(current_filter_exp))}
+            None => {Some(current_filter_exp)}
+        }
+    }   
+    combined_filter_exp
+}
+
+
+/// Computes by linear interpolation the toal and continuum intensity
+/// Input: 
+/// Output: (intens_result,cont_result) where intens is the total intensity; cont is the continuum intensity
+/// 
+pub fn interpolate(
```

Comment:
IMHO you can make a separate module (e.g. "InterpolatingFunction") that takes a grid to construct it and then behaves like a function so that you can `f(x,y,z)` in a transparant way. This will for sure be more broadly useful than only intensity grids.